### PR TITLE
DCOS-13451: Simplify the runtime configuration

### PR DESCRIPTION
**Pull request by wavesoft**
This PR replaces #1869 since it's base branch got completely out of sync with master.

new-run-service mov

**Checklist**
- [x] Did you add a JIRA issue in a commit message or as part of the branch name?
- [x] Did you add new unit tests?
- [x] Did you add new integration tests?

### CODE REVIEW

**Comment by jfurrow** [comment]
Overall the code looks good. I'm noticing a problem where the errors that are displayed inline in the form are out-of-sync with the list of errors at the top. It seems some of the validation occurs on blur events and others occur on change events.

See the following gif:


I also think the orange `Experimental` warning looks weird — is this how it looks in the mockup?

**Comment by leemunroe** [comment]
@wavesoft Nice.

1. Is there a reason we went with an orange experimental label over blue? [Mockup](https://cl.ly/2m3C0m253f3K)

2. What was the thinking behind going with "More Settings" over "Advanced Settings"?

**Comment by wavesoft** [comment]
@jfurrow :
* When you start editing a field it's errors are muted. You might have a point here and avoid removing muted errors from the global error section (that causes this nasty shifting of the contents)
* When you blur, the actual validation takes place. That's why it feels that some validation occurs on blur.

@leemunroe 
* Unfortunately `cnvs` does not have a blue badge and I wanted to keep some consistency and not introduce another badge outside the library. @ashenden shall we add a blue badge to `cnvs`? So my options were either green or orange. Even though green looked better, it conveyed the wrong message, so we decided to go with orange since this feature is experimental.
* I didn't like the fact that the "Grant Runtime Privileges", "Force pull" and the other options seemed grouped under the "Placement Constraints" header, so I added an additional header, "Advanced Settings", which kind of makes sense. However, this made the "Advanced Settings" collapsible redundant, so I changed it in "More Settings". If you think that's wrong I can easily change it back.

**Comment by ashenden** [comment]
@wavesoft sounds like a wise addition.  I won't be able to get the update in CNVS for a while.  If you'd like to submit a PR, it should be simple enough and @jfurrow or I can probably merge it in quickly.

**Comment by jfurrow** [comment]
@wavesoft Thanks for the detailed explanation! In any case, my concern is unrelated to this PR, so it gets my label.

**Comment by leemunroe** [comment]
@wavesoft Gotcha. Heading makes sense. I have some spacing concerns but I'll create a separate issue for this.

I'd still like to see the experimental badge blue and the `More information` link beside Mesos Runtime removed. I think both this elements currently draw too much attention.

**Comment by wavesoft** [comment]
@leemunroe I removed the link and changed the class of the badge to `badge-info`. Currently that's missing from `cnvs` and it appears grey, but when we update it it should appear blue.

image

**Comment by rcorral** [comment]
@wavesoft sorry to come in late here, but could you please add integration tests?

**Comment by wavesoft** [comment]
@rcorral added 👍

**Comment by wavesoft** [comment]
run integration tests

**Comment by wavesoft** [comment]
run unit tests

**Comment by wavesoft** [comment]
run unit tests

**Comment by wavesoft** [comment]
run unit tests

**Comment by nLight** [comment]
run unit tests

**Comment by nLight** [comment]
Feel free to merge as they pass...

**Comment by nLight** [comment]
@wavesoft would you please rebase this one to see if the build error goes away?

**Comment by nLight** [comment]
run unit tests

**Comment by nLight** [comment]
run unit tests

**Comment by nLight** [comment]
run unit tests

File: tests/pages/services/ServiceFormModal-cy.js [hunk 1]
```diff
@@ -240,4 +240,190 @@ describe('Service Form Modal', function () {
...
+            'Single Container',
+            'Multi-container (Pod)',
+            'JSON Configuration'
+          ]);
+        });
+    });
+
+  });
+
+  context.only('Create Layout (Single Container)', function () {
```
Comment (orlandohohmeier) [inline tests/pages/services/ServiceFormModal-cy.js]: Please remove the `only` to make sure that all test run.

**Comment by nLight** [comment]
run integration tests

**Pull request merged**